tvtk/tests/test_array_handler.py

```diff
@@ -396,6 +396,20 @@ def test_array_cache(self):
         del varr
         self.assertEqual(len(cache), 0)
 
+    def test_vtk2array_appended_array(self):
+        """Test the vtk2array can tolerate appending a cached array."""
+        # array is cached upon array2vtk is called
+        arr  = numpy.arange(8).reshape(2, 4)
+        vtk_arr = array_handler.array2vtk(arr)
+        arr1 = array_handler.vtk2array(vtk_arr)
+
+        # the vtk array is appended, shapes don't match cached array anymore
+        vtk_arr.InsertTuple4(2, 1., 2., 3., 4.)
+
+        # arr2 has a different shape
+        arr2 = array_handler.vtk2array(vtk_arr)
+        self.assertEqual(arr2.shape, (3, 4))
```

Comment:
you should also check the values of the array to make sure that all is as expected